hw/ip/otbn/dv/otbnsim/sim/isa.py

```diff
@@ -108,6 +108,11 @@ def __init__(self, op_vals: Dict[str, int]):
         self.grs1 = op_vals['grs1']
         self.imm = op_vals['imm']
 
+        # The bits of self.imm, sign-extended from 12 to 32 bits and then
+        # regarded as an unsigned number. This is used for bitwise
+        # register-immediate instructions (ANDI, OR, XORI)
+        self.unsigned_imm = (1 << 32) + self.imm if self.imm < 0 else self.imm
```

Comment:
That's the kind of code that should go into the decoding section of each instruction: how to interpret bits in the instruction.

Since we don't have that, could we please make it explicit that there is an interpretation happening by providing this functionality as helper, e.g. `sign_extend()`? With that, you could write `val2 = sign_extend(self.imm)` in the execute section of the instruction, making it clear what is happening.

Reply:
Hmm, not sure about the decoding section. But you're right about making things more explicit. Since the sign extension has already happened (my comment above wasn't quite right: the value is a signed integer at this point), I'll call an `as_u32` function: that should hopefully be a clear name.